### PR TITLE
Fix master lost file count metric is always 0

### LINKED ISSUE #14400: Metric MASTER_LOST_FILE_COUNT is not working correctly
**Alluxio Version:**
master

**Describe the bug**

Metric `MASTER_LOST_FILE_COUNT` should show the number of lost files, instead it is always 0.

**To Reproduce**

1. Set property `alluxio.master.lost.worker.file.detection.interval=30s` so the bug can be observed more quickly.
2. Launch a clean Alluxio cluster.
3. Create a file in Alluxio filesystem, e.g. 
    ```console
    bin/alluxio fs -Dalluxio.user.file.writetype.default=MUST_CACHE -Dalluxio.user.block.size.bytes.default=128KB copyFromLocal logs/master.log /master.log
    ```
4. Kill the worker process, clear its storage, by running `bin/alluxio formatWorker`.
5. Restart the worker, wait until it registers with the master.
6. Observe the metrics, by 
    ```console
    curl 127.0.0.1:19999/metrics/json/ | jq '.gauges | [."Master.LostFileCount", ."Master.LostBlockCount"]'
    ```

**Expected behavior**
Both metrics, `Master.LostBlockCount` and `Master.LostFileCount` are not 0.

**Observed Behavior**
`Master.LostFileCount` is 0, while `Master.LostBlockCount` is not.

image

**Urgency**
Urgent. Blocker for 2.7 release.

**Are you planning to fix it**
See the below context.

**Additional context**
I did some preliminary debugging:

1. Set the gauge `MASTER_LOST_FILE_COUNT` from a cached one to a not cached one.
2. Insert tracing loggings to `alluxio.master.file.DefaultFileSystemMaster#getLostFiles` and `alluxio.master.block.DefaultBlockMaster#getLostBlocksCount`

The logging shows that both methods get called when retrieving the metrics, and `getLostFiles` returned a non-empty list, but the metric ended up showing as 0.

image

### PR DESCRIPTION
**Pull request by dbw9580**
Fix #14400, and tweak metric description.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public LostFileDetector(FileSystemMaster fileSystemMaster, InodeTree inodeTree)
     mFileSystemMaster = fileSystemMaster;
     mInodeTree = inodeTree;
     MetricsSystem.registerCachedGaugeIfAbsent(MetricKey.MASTER_LOST_FILE_COUNT.getName(),
-        mFileSystemMaster.getLostFiles()::size);
+        () -> mFileSystemMaster.getLostFiles().size());
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/file/LostFileDetector.java:48]: duh

**Review by ggezer — APPROVED** [review]
Thanks for finding and fixing this!

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.7 please

**Pull request merged**